diff --git a/src/TrackerWatcherCommand.cc b/src/TrackerWatcherCommand.cc
index 960a377e..cc8edd19 100644
--- a/src/TrackerWatcherCommand.cc
+++ b/src/TrackerWatcherCommand.cc
@@ -240,15 +240,11 @@ bool TrackerWatcherCommand::execute()
     // will get Segmentation fault.
     if (trackerRequest_->success()) {
       if (trackerRequest_->processResponse(btAnnounce_)) {
-        btAnnounce_->announceSuccess();
-        btAnnounce_->resetAnnounce();
         addConnection();
       }
-      else {
-        btAnnounce_->announceFailure();
-        if (btAnnounce_->isAllAnnounceFailed()) {
-          btAnnounce_->resetAnnounce();
-        }
+      btAnnounce_->announceFailure();
+      if (btAnnounce_->isAllAnnounceFailed()) {
+        btAnnounce_->resetAnnounce();
       }
       trackerRequest_.reset();
     }
